### ISSUE TITLE
Prevent copying Magento code but inherit from parent class instead.

### ISSUE THREAD

**Issue by ericclaeren** [translations-com/glc-m2#2]
Hi,

Our code is clashing for yours on the Store view edit form.

`<preference for="Magento\Backend\Block\System\Store\Edit\Form\Store" type="TransPerfect\GlobalLink\Block\System\Store\Edit\Form\Store" />`

Which isn't a really big problem, we could override your class with a preference and add our code. But I noticed that the complete code of Magento is copied, which may causes troubles in the future if Magento would decide to add another field.

Would you mind inheriting Magento fields from it's parent class instead of a hard copy.

E.g.
```
protected function _prepareStoreFieldset(Form $form)
{
    parent::_prepareStoreFieldset($form);

    $storeModel = $this->_coreRegistry->registry('store_data');
    $fieldset   = $form->getElement('store_fieldset');

    $fieldset->addField(
        ... your field ...
    );
}
```

**Comment by justin7griffin** [comment]
Hi Eric,

We will consider modifying this for our next release. Thanks for pointing this out!

Best regards,
Justin Griffin

**Comment by ericclaeren** [comment]
Hi @justin7griffin 

I think I have a better suggestion. We had a custom preference on the same class and our class was higher in order and therefor overruled your addition of the locale. Therefor I missed the field in the Form UI.

What I did was rewrite our custom field to an Observer instead of the preference based on this event: adminhtml_store_edit_form_prepare_form

And the code was much cleaner:

```
/**
 * @param Observer $observer
 */
public function execute(Observer $observer)
{
    /** @var Store $block */
    $block = $observer->getEvent()->getBlock();

    $form       = $block->getForm();
    $fieldset   = $form->getElement('store_fieldset');
    $storeModel = $this->registry->registry('store_data');

    $fieldset->addField(
        .... field settings ....
    );
}
```

Now both are working, if you would go for a similar solution the change of conflicts with preferences would be less. Ofcourse I don't know if there are issues with older versions of Magento. This works for our current 2.3.5 version.

**Comment by justin7griffin** [comment]
I don't anticipate there would be any issue with any of the versions of Magento we're worried about, and 2.3.x will be EOL soon anyways. What are you using to trigger the observer?

**Comment by ericclaeren** [comment]
> I don't anticipate there would be any issue with any of the versions of Magento we're worried about, and 2.3.x will be EOL soon anyways. What are you using to trigger the observer?

adminhtml_store_edit_form_prepare_form as stated in my comment ;)

2.3.x won't be EOL for a considerable time. See https://devdocs.magento.com/release/

**Comment by justin7griffin** [comment]
Sorry, missed that part! I could've sworn I saw a roadmap where 2.3.x was ending much sooner. We usually recommend updating past 2.3.5 anyways, due to the number of issues with emails that existed in early 2.3.x versions.

**Comment by justin7griffin** [comment]
This will be included starting in version 1.8.4_RC1. We don't currently have a release date set, as we are waiting for Magento 2.4.2's early access to verify there are no other issues that need working on.
